infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/KubernetesServerExposer.java

```diff
@@ -129,35 +134,68 @@ public KubernetesServerExposer(
    * @param servers servers to expose
    * @see UniqueNamesProvisioner#provision(KubernetesEnvironment, RuntimeIdentity)
    */
-  public void expose(Map<String, ? extends ServerConfig> servers) {
+  public void expose(Map<String, ? extends ServerConfig> servers) throws InfrastructureException {
     Map<String, ServerConfig> internalServers = new HashMap<>();
     Map<String, ServerConfig> externalServers = new HashMap<>();
+    Map<String, ServerConfig> secureServers = new HashMap<>();
 
     servers.forEach(
         (key, value) -> {
           if ("true".equals(value.getAttributes().get(INTERNAL_SERVER_ATTRIBUTE))) {
+            // Server is internal. It doesn't make sense to make an it secure since
+            // it is available only within workspace servers
             internalServers.put(key, value);
           } else {
-            externalServers.put(key, value);
+            // Server is external. Check if it should be secure or not
+            if ("true".equals(value.getAttributes().get(ServerConfig.SECURE_SERVER_ATTRIBUTE))) {
```

Comment:
Boolean.parseBoolean(value.getAttributes().get(ServerConfig.SECURE_SERVER_ATTRIBUTE)) ?

Reply:
Thanks for pointing. And it was implemented in this way for internal servers. I think it is done in this way because comparing with `true` obviously shows that only this value is suitable here. Also, it is documented [here](https://github.com/eclipse/che/blob/2c2bd6637c8169018e2b33397f9d738bbf82a0b1/core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/workspace/config/ServerConfig.java#L26-L27) that only `true` will be considered as right value for internal/secure servers. While `parseBoolean` checks value with `equalsIgnoreCase`. I'd leave it as it is, at least for now.